src/main/game/entities/Entity.java

```diff
@@ -16,9 +16,10 @@
     protected MovementManager movementManager;
 
     //TODO Get movementManager into constructor
-    public Entity(EntityId entityId){
+    public Entity(EntityId entityId, MovementManager movementManager){
```

Comment:
Why do structures need a movement manager if they cannot move? Why can we not just tie this to be part of Army and Unit only?

If it does more than just movement, we should probably rename it to something like 'PlacementManager'